### PR TITLE
Lookup index by GENCODE

### PR DESCRIPTION
**Pull request by k-kom**
- [previous ref-gene implementation](https://github.com/chrovis/varity/pull/42) lacks support to look up regions by GENCODE id

### CODE REVIEW

**Comment by federkasten** [comment]
@totakke This PR is continued from #42. I would be happy if you could check it out along with #42. (I accidentally skipped your review there 🙇 )

File: test/varity/ref_gene_test.clj [hunk 1]
```diff
@@ -141,6 +141,32 @@
...
+                            :chr "chr12"}]
+        gencode-idx (rg/index gencode-test-data)]
+    (are [?param ?output]
+        (= (-> (rg/ref-genes ?param gencode-idx) first :name)
+           ?output)
+      "ENST00000449692.3" "ENST00000449692.3"
+      "KRAS" "ENST00000311936.8"
+
+      "FOO" nil
+      "ENSP00000000001.8" nil)))
```
Comment (totakke) [inline test/varity/ref_gene_test.clj:168]: `ref-genes` fn returns multiple entries, so you should make `?output` multiple. In this case, the search by `KRAS` gene returns two entries.

```clojure
    (are [?param ?output]
        (= (map :name (rg/ref-genes ?param gencode-idx))
           ?output)
      "ENST00000449692.3" ["ENST00000449692.3"]
      "KRAS" ["ENST00000311936.8" "ENSP00000000001.8"]

      "FOO" []
      "ENSP00000000001.8" [])
```

**Review by totakke — COMMENTED** [review]
Thank you for the fix and the addition of a test case. I have left a comment on the test case.

File: test/varity/ref_gene_test.clj (hunk 1 above)
Comment (k-kom) [reply test/varity/ref_gene_test.clj]: Thank you for your review. That makes sense.
I updated the test 👍

**Review by totakke — APPROVED** [review]
LGTM

**Comment by federkasten** [comment]
@k-kom @totakke Thanks for the implementation and review. I will merge.

**Pull request merged**